src/main/java/com/imaee/propinq/buildings/data/models/Building.java

```diff
@@ -0,0 +1,52 @@
+package com.imaee.propinq.buildings.data.models;
+
+import com.imaee.propinq.shared.data.models.Image;
+import com.imaee.propinq.properties.data.models.Property;
+import com.imaee.propinq.shared.data.models.Review;
+import jakarta.persistence.*;
+import lombok.*;
+import com.imaee.propinq.shared.data.models.Locatable;
+import com.imaee.propinq.users.data.models.User;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+@Entity(name = "buildings")
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Data
+public class Building extends Locatable {
+
+    @Id
+    private UUID buildingId = UUID.randomUUID();
+
+    @NonNull
+    private String name;
+
+    @NonNull
+    private String description;
+
+    @NonNull
+    private String address;
+
+    @NonNull
+    @ManyToMany(mappedBy = "building", cascade = CascadeType.ALL)
+    private List<Image> images;
```

Comment:
A esto también inicializalo como una `emptyList()`

```suggestion
    private List<Image> images = Collections.emptyList();
```